Handle ngModelOptions

psyho opened the pull request:
As of Angular 1.3, it's possible to use jQuery style getter/setter functions in ngModel in addition to just mutating a field. You can do that by adding [ng-model-options][1]:

```html
<input ng-model="foo" ng-model-options="{getterSetter: true}">
```

where `foo` is a function that can be invoked with either 0 parameters (then it acts as a getter), or 1 parameter (then it acts as a setter).

Because of this, $watching on attrs.ngModel is no longer sufficient. You need to $watch the actual $modelValue on ngModel controller.

In addition to the fix for the aforementioned issue, I also added missing dependencies needed to run tests to the package.json file.

[1]: https://docs.angularjs.org/api/ng/directive/ngModelOptions

frapontillo commented:
Looks fine.
Can you please explain why `ms` and `on-finished` are needed?

psyho commented:
I didn't look into that. I just tried running `npm test` and it failed with message like "Error: Cannot find module '<module name>'", so I installed that module with "npm --save-dev <module name>", rinse and repeat.

frapontillo commented:
I don't think they are needed at all, since the Travis build has always succeeded so far.

psyho commented:
You're right. After removing node_modules I run into some more issues after npm install, so I updated npm/node. After that, the project dependencies install cleanly, and none of the dependencies I added to package.json turned out to be necessary. Sorry for the confusion.
